core-bundle/src/Twig/Global/ContaoVariable.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Contao\CoreBundle\Twig\Global;
+
+use Contao\CoreBundle\Csrf\ContaoCsrfTokenManager;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\Security\Authentication\Token\TokenChecker;
+use Contao\PageModel;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class ContaoVariable
+{
+    public function __construct(
+        private readonly ContaoFramework $framework,
+        private readonly RequestStack $requestStack,
+        private readonly TokenChecker $tokenChecker,
+        private readonly ContaoCsrfTokenManager $tokenManager,
+    ) {
+    }
+
+    public function page(): PageModel|null
+    {
+        $request = $this->requestStack->getCurrentRequest();
+
+        if (null === $request || !$request->attributes->has('pageModel')) {
+            return null;
+        }
+
+        $pageModel = $request->attributes->get('pageModel');
+
+        if ($pageModel instanceof PageModel) {
+            return $pageModel;
+        }
+
+        if (
+            isset($GLOBALS['objPage'])
+            && $GLOBALS['objPage'] instanceof PageModel
+            && (int) $GLOBALS['objPage']->id === (int) $pageModel
+        ) {
+            return $GLOBALS['objPage'];
+        }
+
+        $this->framework->initialize();
+
+        return $this->framework->getAdapter(PageModel::class)->findByPk((int) $pageModel);
+    }
+
+    public function security(): SecurityVariable
+    {
+        return new SecurityVariable($this->tokenChecker);
+    }
+
+    public function requestToken(): string
```

Comment:
```suggestion
    public function request_token(): string
```
Should we name this `request_token` to be compatible with naming conventions in Twig?

Reply:
no, this is a function call, `app.request.requestUri` is uppercase as well. The usage in Twig would be `contao.requestToken` or `contao.security.previewMode` etc.

Reply:
```suggestion
    public function getRequest_token(): string
```
Looks like this in the naming convention to use here? see https://github.com/symfony/symfony/blob/d621936e8c638247de4c5d1cd75cc5ff55b84c3e/src/Symfony/Bridge/Twig/AppVariable.php#L196

Reply:
Yes, Symfony uses

* `app.current_route`
* `app.current_route_parameters`
* `app.enabled_locales`

https://symfony.com/doc/6.4/templates.html#the-app-global-variable

And our previous global variable was also snake case (`request_token`), so I think it should be `contao.request_token` 👍

Reply:
that looks horrible to me …